### PR TITLE
Cleaning and Fixing the User Interface

### PR DESCRIPTION
**Pull request by qchuchu**
On a modifié le code pour qu'il soit plus compréhensible et lisible.

- Models : Contient toute la logique pour l'initialisation de l'interface et l'envoie des commandes
- Heuristics : Chacun peut rajouter sa propre heuristique, en s'aidant du README en entrée, et en modifiant le main

### CODE REVIEW

File: heuristics/h1.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    
+
+def heuristic1(Player, Game):
+    moves_list = []
+    w0,w1,w2 = (1,1,1)
+    opponent_dict = Game.get_opponent_positions()
+    humans_dict = Game.get_humans_positions()
+    for i,group in enumerate(Player.get_species().get_groups()):
+        position_x, position_y = group.get_position()
+        max_value = -100000
```
Comment (hinosxz) [inline heuristics/h1.py:37]: nit: Prefer `math.inf` or `-float("inf")`

File: heuristics/h1.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    y_init -= nb_coups_diagonale * sign(y_init - y_final)
+    nb_coups_ligne = max(abs(x_init - x_final), abs(y_init - y_final))
+    return nb_coups_diagonale + nb_coups_ligne
+    
+
+def heuristic1(Player, Game):
+    moves_list = []
+    w0,w1,w2 = (1,1,1)
+    opponent_dict = Game.get_opponent_positions()
+    humans_dict = Game.get_humans_positions()
```
Comment (hinosxz) [inline heuristics/h1.py]: I guess you assumed we always play humans?

File: heuristics/h1.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    for i,group in enumerate(Player.get_species().get_groups()):
+        position_x, position_y = group.get_position()
+        max_value = -100000
+        max_move = (0,0)
+        for move in itertools.product([-1,0,1],[-1,0,1]):
+            value = 0
+            if move[0]!=0 or move[1]!=0:
+                group.move([position_x + move[0], position_y + move[1]])
+                d_0, n_0 = group.get_position(), group.get_size()
```
Comment (hinosxz) [inline heuristics/h1.py:43]: nit: Can we choose better than `d` and `n` as variable names. They're not super explicit

File: main.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+parser = argparse.ArgumentParser()
+parser.add_argument('--name', '-N', help="Name of the player", type=str, default='JPP2')
+parser.add_argument('--host', '-H', help="the host to use", type=str, default='127.0.0.1')
+parser.add_argument('--port', '-P', help="the port to use", type=int, default=5555)
+args = parser.parse_args()
+
+if __name__ == "__main__":
+    Game = Game(args.host, args.port)
+    Player = Player(args.name, Game)
+    print("Vous êtes les {}".format(Game._type))
```
Comment (hinosxz) [inline main.py]: 🇫🇷

File: main.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    while Game.is_running:
+        Game.update_map()
+        if Game.is_running:
+            moves_dict = heuristic1(Player, Game)
+            print(moves_dict)
```
Comment (hinosxz) [inline main.py]: Why do we print something here?

File: models/game.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+            for j, cell in enumerate(row):
+                if cell[0] != 0:
+                    humans_dict[(i,j)] = cell[0]
+        print(humans_dict)
+        return humans_dict
+
+    def get_map(self):
+        return self._map
+
+    def get_opponent_positions(self):
```
Comment (hinosxz) [inline models/game.py:42]: These methods are more related to the board than the game itself. I wonder if we could have a `Board` class with all the map related stuff and keep only the methods interacting with the server in the `Game`. Tell me what you think 🙂

File: models/species.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        ----------
+        coordinates : list
+            list de taille 2 contenant les coordonnées du groupe souhaité
+        """
+        for group in self.get_groups():
+            if group.get_position() == coordinates:
+                return group
+        raise ValueError('No group is on the position ({},{})'.format(x,y))
+
+    # def divise(self, group_1, nb1, nb2):
```
Comment (hinosxz) [inline models/species.py]: Why is this commented out?

File: .gitignore [hunk 8]
```diff
@@ -1,3 +1,5 @@
+old_main.py
```
Comment (hinosxz) [inline .gitignore:1]: I believe you can remove this

**Review by hinosxz — COMMENTED** [review]
Looks super nice overall 👏  I only left a few code-related comments. Also, I let's rebase this branch off of `master` once `black` has been enabled (https://github.com/hinosxz/tw-AI-light/pull/1).

One other thing is I guess we just haven't talked about it but since we're pushing this on GitHub I think we should write the code in English.

File: heuristics/h1.py (hunk 1 above)
Comment (qchuchu) [reply heuristics/h1.py:37]: @Barth2501

File: heuristics/h1.py (hunk 2 above)
Comment (qchuchu) [reply heuristics/h1.py]: No actually in the heuristic @Barth2501 was principally interested in looking at the opponent (werewolves or vampire depending on the type), and the humans positions. The humans are not a type that we can play

File: heuristics/h1.py (hunk 3 above)
Comment (qchuchu) [reply heuristics/h1.py:43]: @Barth2501 told you ;)

File: main.py (hunk 5 above)
Comment (qchuchu) [reply main.py]: The print was just for debugging, but clearly it will be moved out after

File: models/species.py (hunk 7 above)
Comment (qchuchu) [reply models/species.py]: Because those functions were written by @Barth2501 but not tested and need to be implemented in the future.

**Comment by qchuchu** [comment]
Thanks @hinosxz for your feedbacks. I will modify the comments (as I'm the one who did them) in English.

But yeah for the next features let's do everything else in english.

File: heuristics/h1.py (hunk 2 above)
Comment (hinosxz) [reply heuristics/h1.py]: Oops, my bad I forgot for a moment we cannot play humans.

**Review by hinosxz — APPROVED** [review]
Looks definitely better than before. Let's iterate on that 👍

**Comment by qchuchu** [comment]
Thanks Gauthier ! See you on Friday to do some sale

**Comment by Barth2501** [comment]
Yes let’s meet at Gif-sur-Yvette this time!

________________________________
De : Quentin Churet <notifications@github.com>
Envoyé : Wednesday, February 26, 2020 1:44:58 PM
À : hinosxz/tw-AI-light <tw-AI-light@noreply.github.com>
Cc : Barthélemy Lancelot <barthelemy.lancelot@student.ecp.fr>; Mention <mention@noreply.github.com>
Objet : Re: [hinosxz/tw-AI-light] Cleaning and Fixing the User Interface (#3)

Thanks Gauthier ! See you on Friday to do some sale

**Pull request merged**